background.js

```diff
@@ -1,126 +1,133 @@
-chrome.storage.local.get('MB', function (obj) {
-    var mem_bytes;
-    if (obj.MB === undefined) {
-        mem_bytes = 32*1E6;
+chrome.storage.local.get(['DecoderMemoryLimitInMegabytes','DebugOutput'], function (obj) {
+    var DecoderMemoryLimitInbytes;
+    if (obj.DecoderMemoryLimitInMegabytes === undefined) {
+        DecoderMemoryLimitInbytes = 32*1E6;
     } else {
-        mem_bytes = obj.MB*1E6 ;
+        DecoderMemoryLimitInbytes = obj.DecoderMemoryLimitInMegabytes*1E6 ;
     }
-    Tiff.initialize({ TOTAL_MEMORY: mem_bytes });
-});
+    Tiff.initialize({ TOTAL_MEMORY: DecoderMemoryLimitInbytes });
 
-function convert_binary_string_to_buffer( data ){
-  var buffer, view, a_byte;
-
-  buffer = new ArrayBuffer( data.length );
-  view = new DataView( buffer );
-
-  data.split( '' ).forEach( function( c, i ){
-    a_byte = c.charCodeAt();
-    view.setUint8( i, a_byte & 0xff );
-  });
-
-  return view.buffer;
-}
-
-function create_request_for_url( url ){
-  var request;
-
-  console.log( 'Starting XHR request for', url );
-  request = new XMLHttpRequest();
-  request.open( 'GET', url, false );
-  request.overrideMimeType( 'text\/plain; charset=x-user-defined' );
-  request.send( null );
-  console.log( 'Finished XHR request' );
+    // Disable console.log if DebugOutput is false
+    if (obj.DebugOutput === undefined) {
```

Comment:
Did you consider just doing:

    if( options.DebugOutput !== true ){
      console.log = function(){};
    }

Right now it would be on for anything that isn't `undefined` or `false`.